presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -190,11 +190,51 @@ public void testBasicTypes()
                 .addRoundTrip("integer", "123456789", INTEGER)
                 .addRoundTrip("smallint", "32456", SMALLINT, "SMALLINT '32456'")
                 .addRoundTrip("tinyint", "5", SMALLINT, "SMALLINT '5'")
-                .addRoundTrip("double", "123.45", DOUBLE, "DOUBLE '123.45'")
-                .addRoundTrip("real", "123.45", REAL, "REAL '123.45'")
                 .execute(getQueryRunner(), prestoCreateAsSelect("test_basic_types"));
     }
 
+    @Test
+    public void testReal()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("real", "NULL", REAL, "CAST(NULL AS real)")
+                .addRoundTrip("real", "3.14", REAL, "REAL '3.14'")
+                .addRoundTrip("real", "3.1415927", REAL, "REAL '3.1415927'")
+                .addRoundTrip("real", "'NaN'::real", REAL, "CAST(nan() AS real)")
+                .addRoundTrip("real", "'-Infinity'::real", REAL, "CAST(-infinity() AS real)")
+                .addRoundTrip("real", "'+Infinity'::real", REAL, "CAST(+infinity() AS real)")
+                .execute(getQueryRunner(), postgresCreateAndInsert("tpch.postgresql_test_real"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("real", "NULL", REAL, "CAST(NULL AS real)")
+                .addRoundTrip("real", "3.14", REAL, "REAL '3.14'")
+                .addRoundTrip("real", "3.1415927", REAL, "REAL '3.1415927'")
+                .addRoundTrip("real", "nan()", REAL, "CAST(nan() AS real)")
+                .addRoundTrip("real", "-infinity()", REAL, "CAST(-infinity() AS real)")
+                .addRoundTrip("real", "+infinity()", REAL, "CAST(+infinity() AS real)")
+                .execute(getQueryRunner(), prestoCreateAsSelect("presto_test_real"));
+    }
+
+    @Test
+    public void testDouble()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("double precision", "NULL", DOUBLE, "CAST(NULL AS double)")
+                .addRoundTrip("double precision", "1.0E100", DOUBLE, "1.0E100")
+                .addRoundTrip("double precision", "'NaN'::double precision", DOUBLE, "nan()")
+                .addRoundTrip("double precision", "'+Infinity'::double precision", DOUBLE, "+infinity()")
```

Comment:
nit: Any specific reason to use the PostgreSQL specific CAST syntax?

Reply:
Pre-existing. Anyway, the postgresql-specific syntax is heavily advertised in postgresql docs, so why not? The expression as a whole remains postgresql-specific anyway.